src/Commands/IndexCategoriesCommand.php

```diff
@@ -25,11 +25,11 @@ public function handle(): int
 
     public function getCategories()
     {
-        return config('rapidez.models.category')::query()
-            ->select((new (config('rapidez.models.category')))->qualifyColumns(['entity_id', 'name', 'url_path', 'children_count']))
+        return config('rapidez.models.category')::select((new (config('rapidez.models.category')))->qualifyColumns(['entity_id', 'name', 'url_path']))
+            ->withCount('productIndices')
             ->whereNotNull('url_key')
             ->whereNot('url_key', 'default-category')
-            ->where('children_count', '>', 0)
+            ->having('product_indices_count', '>', 0)
```

Comment:
In this case you probably do not have to go this complex.
By simply querying a relationships existence Laravel already creates a good query for you
https://laravel.com/docs/10.x/eloquent-relationships#querying-relationship-existence

Where it uses subselects in a single query, we do not need to know the count. Only that there is at least 1 visible product.

Considering you've already added your scope for visibility 
`->has('products')` should suffice.

Do be sure to dump the query just to make sure it gets built into a good query

Reply:
Ah, didn't know about this one. Looks like it works great & it's efficient enough that the `productIndices` relation is probably no longer necessary :)